### PR TITLE
Improve android build time and assembly size

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Refactoring (no functional changes, no api changes)

## What is the new behavior?

The Uno.UI build now skips the creation of android resources designer classes, which are not used by uno itself (or in very rare occasions), therefore making the build significantly faster, but also reduces the size of the app and provides better startup time.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -72,11 +72,19 @@
     Force the compilation using Roslyn Generators on net461 (Unit Tests)
     -->
 	  <UnoUIUseRoslynSourceGenerators Condition="'$(TargetFramework)'=='net461'">true</UnoUIUseRoslynSourceGenerators>
-	  
-    <!--
+
+	  <!--
+    Force the compilation using Roslyn Generators on monoandroid
+    -->
+	  <UnoUIUseRoslynSourceGenerators Condition="'$(TargetFramework)'=='monoandroid11.0' or '$(TargetFramework)'=='monoandroid12.0' or '$(TargetFramework)'=='xamarinmac20'">true</UnoUIUseRoslynSourceGenerators>
```
Comment (spouliot) [inline src/Directory.Build.props:79]: `or '$(TargetFramework)'=='xamarinmac20'`
^ that does not match the above comment

Comment (jeromelaban) [reply src/Directory.Build.props:79]: Indeed, I'm mostly trying out options until the property can unconditionally be set to true 🙂

File: doc/articles/uno-development/building-uno-ui.md [hunk 2]
```diff
@@ -122,3 +122,28 @@ To run the synchronization tool:
...
+// aapt resource value: 0x7F010000
+public static int foo = 2130771968;
+```
+
+4. Copy this field to the `Resource.designer.cs` checked into source
+   control, such as: `src/Uno.UI/Resources/Resource.designer.g.Android.cs`
+
+5. Restore the commented code in `Directory.Build.targets`.
+
+_This performance optimization is inspired from @jonathanpeppers's performance work done in https://github.com/dotnet/maui/pull/2606. Thanks Jonathan!_
```
Comment (agneszitte-nventive) [inline doc/articles/uno-development/building-uno-ui.md:149]:
```suggestion
_This performance optimization is inspired by @jonathanpeppers' performance work done in https://github.com/dotnet/maui/pull/2606. Thanks, Jonathan!_
```

**Review by agneszitte-nventive — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

File: doc/articles/uno-development/building-uno-ui.md (hunk 2 above)
Comment (jeromelaban) [reply doc/articles/uno-development/building-uno-ui.md:149]: Will fix in next perf PR, thanks!

**Pull request merged**